tests/unit/test-util/test.submit-addon.js

```diff
@@ -0,0 +1,546 @@
+/* @flow */
+import { promises as fsPromises, readFileSync } from 'fs';
+import path from 'path';
+
+import { fs } from 'mz';
+import { assert, expect } from 'chai';
+import { afterEach, before, beforeEach, describe, it } from 'mocha';
+import * as sinon from 'sinon';
+import nock from 'nock';
+import { File, FormData } from 'node-fetch';
+
+import Client, { signAddon } from '../../../src/util/submit-addon.js';
+import { makeSureItFails } from '../helpers.js';
+import { withTempDir } from '../../../src/util/temp-dir.js';
+
+describe('util.submit-addon', () => {
+
+  describe('signAddon', () => {
+    let statStub;
+    let postNewAddonStub;
+    let putVersionStub;
+
+    beforeEach(() => {
+      statStub = sinon.stub(fsPromises, 'stat')
+        .resolves({isFile: () => true });
+      postNewAddonStub = sinon.stub(Client.prototype, 'postNewAddon');
+      putVersionStub = sinon.stub(Client.prototype, 'putVersion');
+    });
+
+    afterEach(() => {
+      statStub.restore();
+      postNewAddonStub.restore();
+      putVersionStub.restore();
+    });
+
+    const signAddonDefaults = {
+      apiKey: 'some-key',
+      apiSecret: 'ffff',
+      apiHost: 'https://some.url',
+      timeout: 1,
+      downloadDir: '/some-dir/',
+      xpiPath: '/some.xpi',
+      channel: 'some-channel',
+    };
+
+    it.skip('creates Client with parameters', async () => {
+      const apiKey = 'fooKey';
+      const apiSecret = '4321';
+      const apiHost = 'fooPrefix';
+      const downloadDir = '/foo';
+      const clientSpy = sinon.spy(Client);
+
+      await signAddon({
+        ...signAddonDefaults,
+        apiKey,
+        apiSecret,
+        apiHost,
+        downloadDir,
+      });
+
+      sinon.assert.called(clientSpy);
+
+    });
+
+    it('calls postNewAddon if `id` is undefined', async () => {
+      const xpiPath = 'this/path/xpi.xpi';
+      const channel = 'thisChannel';
+      await signAddon({
+        ...signAddonDefaults,
+        xpiPath,
+        channel,
+      });
+      sinon.assert.notCalled(putVersionStub);
+      sinon.assert.calledWith(postNewAddonStub, xpiPath, channel, {});
+    });
+
+    it('calls putVersion if `id` is defined', async () => {
+      const xpiPath = 'this/path/xpi.xpi';
+      const channel = 'thisChannel';
+      const id = '@thisID';
+      await signAddon({
+        ...signAddonDefaults,
+        xpiPath,
+        channel,
+        id,
+      }).then(() => {});
+      sinon.assert.notCalled(postNewAddonStub);
+      sinon.assert.calledWith(putVersionStub, xpiPath, channel, id, {});
+    });
+
+    it('throws error if xpiPath is invalid', () => {
+      statStub.restore();
+      return signAddon(signAddonDefaults)
+        .then(makeSureItFails())
```

Comment:
@eviljeff would you mind to tweak the new tests to not use `makeSureItFails` (and instead use the assertions provided by `"chai-as-promised"`, the `assert(...).isRejected` as in my other review comments related to one of the currently skipped tests).

Reply:
I changed all the tests to use `assert.isRejected` rather than `makeSureItFails` in test.submit-addon.js and `test.sign.js` so neither file imports that function any longer.